### PR TITLE
gui quit on service quit

### PR DESCRIPTION
**Pull request by chrisnojima**
fixes flow issues and listens to the new rpc

@keybase/react-hackers

### CODE REVIEW

**Comment by MarcoPolo** [comment]
:shipit:

File: shared/native/notification-listeners.desktop.js [hunk 1]
```diff
@@ -45,5 +46,9 @@ export default function (dispatch: Dispatch, getState: () => Object, notify: any
         dispatch(favoriteList())
       }
     },
+    'keybase.1.NotifyService.shutdown': () => {
+      console.log('Quitting due to service shutdown')
+      remote.app.quit()
```
Comment (gabriel) [inline shared/native/notification-listeners.desktop.js]: Notifications come on the renderer process?

**Comment by gabriel** [comment]
So this might not do anything for macOS cause we terminate the app process from `ctl stop` (see https://github.com/keybase/client/blob/master/go/client/cmd_ctl_stop_osx.go#L77).

But this is good for linux/windows, so we might want to add a comment in here to that effect.

File: shared/native/notification-listeners.desktop.js (hunk 1 above)
Comment (chrisnojima) [inline shared/native/notification-listeners.desktop.js]: everything comes into the renderer process basically

**Pull request merged**